MyCryptoHQ/MyCrypto
# Add nonce auto-update on transaction send

Closes #2311

### Description

The point of this PR is to make the multi-transaction process easier and more-intuitive

### Changes

* Fix bug with adding transactions to recent transactions local state.
* Extend Recent Transactions to include nonce in local state.
* Changed the state to now empty the transaction state on transaction_broadcast_success. After transaction state is emptied, automatically update nonce to allow for back-to-back tx.
* Changed method for calculating nonce from just `getTransactionCount` to pick the maximum between `getTransactionCount` and highest nonce found for that `from` address in the local state's recent transactions.

### Steps to Test

1. Access wallet
2. Send tx to yourself. Notice that after the transaction success notification appears, the `address`, `gas_price` and `amount` fields reset.  
3. The nonce will automatically refresh to be incremented by one.
